language/vm/src/serializer.rs

```diff
@@ -404,9 +404,12 @@ fn serialize_function_definition(
     function_definition: &FunctionDefinition,
 ) -> Result<()> {
     write_u16_as_uleb128(binary, function_definition.function.0)?;
-    binary.push(function_definition.flags)?;
+    binary.push(function_definition.get_flag_for_serialization())?;
```

Comment:
not sure if I would have made a function in file format or make the flag check in here.
Mention just to make you think about it